Explorer/Assets/DCL/Multiplayer/SDK/Systems/PlayerComponentsHandlerSystem.cs

```diff
@@ -0,0 +1,161 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using CRDT;
+using CrdtEcsBridge.Components;
+using DCL.Character.Components;
+using DCL.Diagnostics;
+using DCL.Multiplayer.Profiles.Systems;
+using DCL.Multiplayer.SDK.Components;
+using DCL.Profiles;
+using ECS.Abstract;
+using ECS.LifeCycle.Components;
+using ECS.SceneLifeCycle;
+using SceneRunner.Scene;
+using UnityEngine;
+using Utility;
+
+namespace DCL.Multiplayer.SDK.Systems
+{
+    // Currently implemented to track reserved entities only on the CURRENT SCENE
+    [UpdateInGroup(typeof(PresentationSystemGroup))]
+    [UpdateAfter(typeof(MultiplayerProfilesSystem))]
+    [LogCategory(ReportCategory.MULTIPLAYER_SDK_COMPONENTS_HANDLER)]
+    public partial class PlayerComponentsHandlerSystem : BaseUnityLoopSystem
+    {
+        private readonly IScenesCache scenesCache;
+        private readonly bool[] reservedEntities = new bool[SpecialEntitiesID.OTHER_PLAYER_ENTITIES_TO - SpecialEntitiesID.OTHER_PLAYER_ENTITIES_FROM];
+        private int currentReservedEntitiesCount;
+
+        public PlayerComponentsHandlerSystem(World world, IScenesCache scenesCache) : base(world)
+        {
+            this.scenesCache = scenesCache;
+            ClearReservedEntities();
+        }
+
+        protected override void Update(float t)
+        {
+            RemovePlayerIdentityDataOnOutsideCurrentSceneQuery(World);
+
+            HandlePlayerDisconnectQuery(World);
+
+            AddPlayerIdentityDataQuery(World);
+        }
+
+        [Query]
+        [None(typeof(PlayerIdentityDataComponent), typeof(DeleteEntityIntention))]
+        private void AddPlayerIdentityData(in Entity entity, ref Profile profile, ref CharacterTransform characterTransform)
```

Comment:
I believe we should filter out own own player? Or not? Because our own one is propagated in `WritePlayerTransformSystem`

Reply:
Uhmm I think you are right, nice catch, we should use the CRDT Entity Id 1 and that way we free up 1 more entity for other player avatars in the scene, I'll update it

Reply:
Tackled and added a tes for that, here https://github.com/decentraland/unity-explorer/pull/632/commits/ed40eac72f871e46aaeb2e1e6cda6fd0097b6234